packages/calypso-e2e/src/lib/components/comments-component.ts

```diff
@@ -28,6 +28,7 @@ export class CommentsComponent extends BaseContainer {
 	async postComment( comment: string ): Promise< void > {
 		// To simulate user action first click on the field. This also exposes the
 		// submit comment button.
+		await this.page.waitForLoadState( 'load' );
```

Comment:
This change looks good, but I wanted to ask a quick design question (mostly for my own clarification!)

In general, do we want to defer `load` waiting downstream to the loaded POM classes, or keep it upstream next to the navigation?

So in this case, the alternative would be updating `PublishedPostsListPage.visitPost` to add the `this.page.waitForLoadState()` and the navigation/click promises resolve.

I think both options are great, I figured I should clarify which approach we wanted to do because I don't think we have that documented anywhere, and then we can replicate that throughout!

Reply:
Ah yes, good question - the answer is that there isn't a specific standard for this within our repo.

Part of the reason for the `BaseContainer` and the presence of a default selector to wait on was to achieve more reliable timing. Through a series of reviews and discussions, Bart and I agreed that it was not the best practice as:
1. POM is now reliant on a CSS class not changing;
2. POM could be waiting unnecessarily if the default wait is too safe.

I believe we agreed in principle that it would be a better approach to have each class method do a cursory check on the state of the page/component before performing actions, but this isn't always followed. Many of these objects (including more recent ones I've written) have methods that jump straight into its actions without first performing any checks.

I think it's a style question. With Playwright's auto-wait, our POMs should be able to fire off action after another and the auto-wait takes care of that for us. In reality though, because of the way calypso has grown over the years, not everything is as consistent as it should be.

My take? I think having additional waits and checks before actions would be beneficial, personally. The default choices in Playwright are good but does not cover all cases and the automatic waits aren't always bulletproof with our product.